File: frontend/lib/models/feed_noticias_model.dart
class FeedNoticia {
  final int? id;
  final int idUsuario;
  final String categoria;
  final String titulo;
  final String descricao;
  final String? fotoCapa;
  final DateTime? criadoEm;
  final DateTime? atualizadoEm;

  FeedNoticia({
    this.id,
    required this.idUsuario,
    required this.categoria,
    required this.titulo,
    required this.descricao,
    this.fotoCapa,
    this.criadoEm,
    this.atualizadoEm,
  });

  factory FeedNoticia.fromJson(Map<String, dynamic> json) {
    return FeedNoticia(
      id: json['ID'],
      idUsuario: json['ID_USUARIO'],
      categoria: json['CATEGORIA'],
      titulo: json['TITULO'],
      descricao: json['DESCRICAO'],
      fotoCapa: json['FOTO_CAPA'],
      criadoEm:
          json['CRIADO_EM'] != null ? DateTime.parse(json['CRIADO_EM']) : null,
      atualizadoEm: json['ATUALIZADO_EM'] != null
          ? DateTime.parse(json['ATUALIZADO_EM'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ID_USUARIO': idUsuario,
      'CATEGORIA': categoria,
      'TITULO': titulo,
      'DESCRICAO': descricao,
      'FOTO_CAPA': fotoCapa,
    };
  }
}

File: backend/models/feedNoticiasModel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../database/db');


const FeedNoticias = sequelize.define('FeedNoticias', {
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
    },
    ID_USUARIO: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
    CATEGORIA: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    TITULO: {
        type: DataTypes.STRING(100),
        allowNull: false,
    },
    DESCRICAO: {
        type: DataTypes.STRING(200),
        allowNull: false,
    },
    FOTO_CAPA: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    // NOVO CAMPO PARA TOKEN DE COMPARTILHAMENTO
    SHARE_TOKEN: {
        type: DataTypes.STRING,
        allowNull: true, // pode ser preenchido quando a notícia for criada ou durante o fluxo de compartilhamento
    },
    CRIADO_EM: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: DataTypes.NOW
    },
    ATUALIZADO_EM: {
        type: DataTypes.DATE,
        allowNull: true,
        defaultValue: DataTypes.NOW
    },
}, {
    tableName: 'FEED_NOTICIAS',
    timestamps: false,
    createdAt: 'CRIADO_EM',
    updatedAt: 'ATUALIZADO_EM'
});

module.exports = FeedNoticias;
